packages/block-library/src/separator/block.json

```diff
@@ -8,16 +8,18 @@
 	"keywords": [ "horizontal-line", "hr", "divider" ],
 	"textdomain": "default",
 	"attributes": {
-		"color": {
-			"type": "string"
-		},
-		"customColor": {
+		"opacity": {
 			"type": "string"
 		}
 	},
 	"supports": {
 		"anchor": true,
-		"align": [ "center", "wide", "full" ]
+		"align": [ "center", "wide", "full" ],
+		"color": {
+			"gradients": true,
+			"background": true,
+			"text": false
```

Comment:
This is interesting. The contrast checker isn't really hitting the mark with its warnings. 

<img width="872" alt="Screen Shot 2022-02-10 at 3 29 36 pm" src="https://user-images.githubusercontent.com/6458278/153337774-fec709b4-20be-4b6a-b436-b1240c5229b7.png">

Maybe we need a way to disable the contrast checker for specific use cases.

Reply:
We've been able to include other support flags in the recent past however they were more generic across blocks e.g. `__experimentalDefaultControls` & `__experimentalSkipSerialization`.

I personally don't have any issues with adding this. I'd suggest we probably prefix it with `__experimental` as well though.

@oandregal Do you have any thoughts or concerns with this approach?

Reply:
Oh, interesting. So it sounds like the contrast checker should be smarter and not report anything if the component has no text color. Does that sound like a good approach?

Reply:
> Oh, interesting. So it sounds like the contrast checker should be smarter and not report anything if the component has no text color. Does that sound like a good approach?

The thing is, it _should_ do that already:

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/contrast-checker/index.js#L38

Clearly it's not though 😆 I'll take a closer look next week to see if there's a component-based solution.

Reply:
> The thing is, it should do that already:

Actually, I've had another play around with the Color Panel and contrast checking. 

The [Color Panel's implementation of the ContrastChecker](https://github.com/WordPress/gutenberg/blob/8675f86f885ca389f3bca39971b22cb897bab61b/packages/block-editor/src/hooks/color-panel.js#L74) is working as (originally) intended in that it attempts to compare the detected background color with **any** detected text color, irrespective of whether the selected block opts into `color.text` or not.

So it's not about whether the component or the selected block has a color, background or otherwise, but whether the selected block's color choices are readable when checked against any _detected_ text color or _detected_ background color.

In the case of the Separator block, even though it technically does not define a color or has any text, a `color` is nevertheless inherited from the `has-background-color` classname, and, if we remove that, from the body.

We should probably be checking for inner text in the [contrast checker's logic](https://github.com/WordPress/gutenberg/blob/8675f86f885ca389f3bca39971b22cb897bab61b/packages/block-editor/src/hooks/color-panel.js#L37). 

```
		if ( !! ref.current.innerText ) {
			setDetectedColor( getComputedStyle( ref.current ).color );
		}

```

This should suit most scenarios. I'll throw up a PR.

I still wonder if it would be helpful to completely disable (or enable) the contrast checker for specific blocks. Maybe a future use case will answer this question for us.

Reply:
> We should probably be checking for inner text in the [contrast checker's logic](https://github.com/WordPress/gutenberg/blob/8675f86f885ca389f3bca39971b22cb897bab61b/packages/block-editor/src/hooks/color-panel.js#L37).

Okay, withdraw that. 

I tried and tested it and it won't work in the editor because most blocks have text: `Add block` or placeholders or whatever 😄 

I'll create an issue and move the conversation there.